webapp/src/components/CreateCompetencyForm.tsx

```diff
@@ -0,0 +1,117 @@
+import { Alert, Card, CardContent, Snackbar, TextField } from '@mui/material';
+import { LoadingButton } from '@mui/lab';
+import React, { FormEventHandler, useState } from 'react';
+
+import { EntityId } from '../types/api';
+
+interface CreateCompetencyFormProps {
+  onCompetencyCreated?: (id: EntityId) => void;
+}
+
+const CreateCompetencyForm = ({
+  onCompetencyCreated,
+}: CreateCompetencyFormProps) => {
+  const [isSavingCompetency, setIsSavingCompetency] = useState(false);
+  const [saveCompetencyError, setSaveCompetencyError] = useState(null);
+  const [isSuccessMessageVisible, setIsSuccessMessageVisible] = useState(false);
+  const [competencyNameEntryText, setCompetencyNameEntryText] = useState('');
+  const [descriptionEntryText, setDescriptionEntryText] = useState('');
+  const [wasCompetencyEntryTextTouched, setWasCompetencyEntryTextTouched] =
+    useState(false);
+  const onSubmit: FormEventHandler = event => {
+    event.preventDefault();
+    setIsSavingCompetency(true);
+    fetch(`${process.env.REACT_APP_API_ORIGIN}/competencies`, {
+      method: 'POST',
+      credentials: 'include',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({
+        label: competencyNameEntryText,
+        description: descriptionEntryText,
+      }),
+    })
+      .then(res => res.json())
+      .then(({ data, error }) => {
+        setIsSavingCompetency(false);
+        if (error) {
+          setSaveCompetencyError(error);
+        }
+        if (data) {
+          setIsSuccessMessageVisible(true);
+          setCompetencyNameEntryText('');
+          setDescriptionEntryText('');
+          if (onCompetencyCreated) {
+            onCompetencyCreated(data.id);
+          }
+        }
+      })
+      .catch(error => {
+        setIsSavingCompetency(false);
+        setSaveCompetencyError(error);
+      });
+  };
+  return (
+    <form onSubmit={onSubmit}>
+      <Card>
+        <CardContent sx={{ pb: 0 }}>
+          <h2>New Competency</h2>
+        </CardContent>
+        <CardContent sx={{ pt: 0 }}>
+          <TextField
```

Comment:
There's a maximum character limit of 255 characters that was defined in the database migration.  Would it be valuable to have some sort of character limit defined in the front-end to prevent users from trying to submit labels that are longer than 255 characters?

Reply:
I think showing them a character limit would be a nice to have feature in the future, for this form and other form fields on the application. For now, when I tried to submit a new competency into the form with a title that's longer than 255 characters, it would not go through and there's a general message that shows up. I believe that will satisfy our issue ticket. 

<img width="761" alt="image" src="https://user-images.githubusercontent.com/16721486/153275236-385640de-b0df-49f2-a93a-bdb2b669a183.png">

Reply:
Thanks for following up!  That's great!  Yes, it'll be good to have some sort of feature in the future, that'll notify the user why it wasn't saved.